libs/blocks/merch-card/merch-card.js

```diff
@@ -0,0 +1,200 @@
+  return descriptionWrapper;
+};
+
+const createTitle = (titles, cardType) => {
+  const titleWrapper = createTag('div', { class: `consonant-${cardType}-title` });
+  titles?.forEach((title) => titleWrapper.appendChild(title));
+  return titleWrapper;
+};
+
+const decorateFooter = (el, altCtaMetaData, styles, cardType) => {
+  const cardFooter = el.querySelector('.consonant-CardFooter');
+  const replacePlaceHolder = async (key, defaultValue) => {
+    const replacedKey = await replaceKey(key, getConfig(), defaultValue);
+    return { replacedKey };
+  };
+  const decorateWithSecureTransactionSign = () => {
+    const secureTransactionWrapper = createTag('div', { class: 'secure-transaction-wrapper' });
+    const label = createTag('span', { class: 'secure-transaction-label' });
+    const secureElement = createTag('span', { class: 'secure-transaction-icon' });
+    secureTransactionWrapper.append(secureElement, label);
+    replacePlaceHolder('secure-transaction').then(({ replacedKey }) => {
+      label.textContent = replacedKey;
+    });
+    return secureTransactionWrapper;
+  };
+
+  const createCheckbox = (checkBoxText) => {
+    cardFooter.querySelector('hr')?.remove();
+    const container = createTag('label', { class: 'checkbox-container' });
+    const input = createTag('input', { id: 'alt-cta', type: 'checkbox' });
+    const checkmark = createTag('span', { class: 'checkmark' });
+    const label = createTag('span', { class: 'checkbox-label' }, checkBoxText.innerHTML);
+    container.append(input, checkmark, label);
+    return container;
+  };
+
+  const decorateAlternativeCta = () => {
+    const altCtaRegex = /href="([^"]*)"/g;
+    if (!altCtaRegex.test(altCtaMetaData[1]?.innerHTML)) return;
+
+    const toggleButtonActiveState = (buttonToAdd, buttonToRemove) => {
+      buttonToAdd.classList.add('button--inactive');
+      buttonToRemove.classList.remove('button--inactive');
+    };
+    const cardFooterRow = el.querySelector('.consonant-CardFooter-row');
+    if ([...el.classList].includes('secure')) {
+      const standardWrapper = createTag('div', { class: 'standard-wrapper' });
+      const secureTransactionWrapper = decorateWithSecureTransactionSign();
+      standardWrapper.append(secureTransactionWrapper, cardFooterRow);
+      cardFooter?.append(standardWrapper);
+    }
+
+    const originalCtaButton = cardFooterRow.querySelector('.consonant-CardFooter-cell--right');
+    const altCtaButton = originalCtaButton.cloneNode(true);
+    const checkboxContainer = createCheckbox(altCtaMetaData[0]);
+
+    altCtaButton.innerHTML = altCtaMetaData[1].innerHTML;
+    altCtaButton.classList.add('button--inactive');
+    checkboxContainer.querySelector('input[type="checkbox"]').addEventListener('change', (event) => {
+      if (event.target.checked) {
+        toggleButtonActiveState(originalCtaButton, altCtaButton);
+      } else {
+        toggleButtonActiveState(altCtaButton, originalCtaButton);
+      }
+    });
```

Comment:
Change event listener can be simplified by inlining logic encapsulated in `toggleButtonActiveState`:
```
checkboxContainer.querySelector('input[type="checkbox"]').addEventListener('change', ({ target: { checked } }) => {
      originalCtaButton.classList.toggle('button--inactive', checked);
      altCtaButton.classList.toggle('button--inactive', !checked);
    });
```

Reply:
done